Solutions/Marain.Workflow.Api.EngineHost.Client/Microsoft/Extensions/DependencyInjection/WorkflowEngineClientServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿// <copyright file="WorkflowEngineClientServiceCollectionExtensions.cs" company="Endjin">
+// Copyright (c) Endjin. All rights reserved.
+// </copyright>
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    using System;
+    using System.Linq;
+    using Corvus.Extensions.Json;
+    using Corvus.Identity.ManagedServiceIdentity.ClientAuthentication;
+    using Marain.Workflows.Client;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Rest;
+
+    /// <summary>
+    /// DI initialization for clients of the workflow engine service.
+    /// </summary>
+    public static class WorkflowEngineClientServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds the workflow engine client to a service collection.
+        /// </summary>
+        /// <param name="services">The service collection.</param>
+        /// <param name="resourceIdForMsiAuthentication">
+        /// The resource id to use when obtaining an authentication token representing the
```

Comment:
I've a horrible feeling I may have written this comment, because this looks very much like code I wrote for other projects...it's not terribly clear what this actually needs to be. The text "representing the hosting service's identity" makes it sound like this is the Managed Identity. It's not. This will be the App ID of the Azure AD Application that was created to secure access to the Workflow Engine's HTTP endpoints (what I think we currently call the ingestion service?). It'll be the App ID that we supply to Easy Auth.

(From a client perspective, it identifies the resource we're telling AAD we want to access. So the way this actually gets used is that we will be saying to AAD "Please give me an access token that lets me use the resource with this ID". And for custom services the resource ID is generally an AppID.)